### Task 1: Building 99% confidence interval using sample standard deviation

Since we do not know the population standard deviation and the sample size is less than 30, we use t-distribution to construct the confidence interval.

In [1]:
import numpy as np

In [2]:
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
n = len(data)

In [3]:
sample_mean = np.mean(data)
sample_std = np.std(data)

In [4]:
confidence_level = 0.99
alpha = 1 - confidence_level

In [5]:
import scipy.stats as stats

In [6]:
t_critical = stats.t.ppf(1 - alpha/2, df=n-1)
margin_of_error_t = t_critical * (sample_std / np.sqrt(n))
confidenceinterval_t = (np.round(sample_mean - margin_of_error_t, 4), np.round(sample_mean + margin_of_error_t, 4))

In [7]:
print(confidenceinterval_t)

(np.float64(1.0952), np.float64(1.3821))


##### 99% confidence interval using t-distribution: (1.0952, 1.3821) million characters

### Building 99% confidence interval using known population standard deviation

In [8]:
population_std = 0.2
z_critical = stats.norm.ppf(1-alpha/2)
margin_of_error_z = z_critical * (population_std / np.sqrt(n))
confidence_interval_z = (np.round(sample_mean - margin_of_error_z, 4), np.round(sample_mean + margin_of_error_z, 4))

In [9]:
print(confidence_interval_z)

(np.float64(1.1057), np.float64(1.3717))


##### 99% confidence interval using z-distribution: (1.1057, 1.3717) million characters